tests/baselines/reference/objectPropertyAsClass.types

```diff
@@ -0,0 +1,69 @@
+=== tests/cases/compiler/index.js ===
+const a1 = {
+>a1 : { foo(): void; }
+>{    foo() {        this.x = 0;    }} : { foo(): void; }
+
+    foo() {
+>foo : () => void
+
+        this.x = 0;
+>this.x = 0 : 0
+>this.x : any
+>this : { foo(): void; }
+>x : any
+>0 : 0
+    }
+}
+
+const a2 = {
+>a2 : { foo: typeof foo; }
+>{    foo: function() {        this.x = 0;    }} : { foo: typeof foo; }
+
+    foo: function() {
+>foo : typeof foo
+>function() {        this.x = 0;    } : typeof foo
+
+        this.x = 0;
+>this.x = 0 : 0
+>this.x : any
+>this : { foo: typeof foo; }
+>x : any
+>0 : 0
+    }
+}
+
+const b1 = {
+>b1 : { foo(): void; }
+>{    /** @class */    foo() {        this.x = 0;    }} : { foo(): void; }
+
+    /** @class */
+    foo() {
+>foo : typeof b1.foo
+
+        this.x = 0;
+>this.x = 0 : 0
+>this.x : any
+>this : { foo(): void; }

```

Comment:
This should say `this : this`, so I screwed something up.

Reply:
Per the PR comments, we're not going to do `@class` and such for property methods, so, resolving.